### PR TITLE
feat: changing svg dimension calculation from puppeteer to resvg

### PR DESCRIPTION
**Pull request by Kreeg**
1. Moving puppeteer to devDependencies.
2. Requiring puppeteer obviously in tests (getting rid of `async-init` and browser manager).
3. Changing reference images in tests

TODO: comparing svg images in tests with resvg for better performance

### CODE REVIEW

**Comment by Kreeg** [comment]
I expected that checks are not running on draft PRs.

**Comment by Kreeg** [comment]
Resvg dimensions differ from puppeteer. And there is no reference because there is no tests with no-dimensions sprites.
Here is what I offer:
1. Saving svgs I've added here
2. Switch back on commit with puppeteer
3. Adding test sprite reference with those svgs and save example
4. Switch back to this branch and run tests to see what is the real difference

**Comment by XhmikosR** [comment]
Whatever makes it easier for you to complete this :)

Assuming we can get rid of puppeteer, and the fewer deps the better, this will be a huge win.

Thanks for working on it!

**Comment by Kreeg** [comment]
@XhmikosR we can't get rid of puppeteer. It takes screenshots from html pages in tests. Maybe there is another solution for this, but I don't know yet

**Comment by XhmikosR** [comment]
If it's only for tests we can certainly live with it, at least for now :)

The important thing is to remove it from production dependencies.

File: package.json
Comment (XhmikosR) [reply package.json:69]: It should be in devDependencies only if puppeteer is needed in tests only.

**Comment by Kreeg** [comment]
I think I new that's wrong with these tests failing.
I'll get rid of async-init soon

File: lib/svg-sprite/calculate-svg-dimensions.js
Comment (XhmikosR) [reply lib/svg-sprite/calculate-svg-dimensions.js:27]: Just a specific error code number when the process exits. We could probably remove this stuff later.

**Comment by Kreeg** [comment]
Conclusions:
1. Puppeteer dimensions differ from phantomjs.
2. RESVG dimensions differ from puppeteer and phantomjs.
3. There is no referencial instrument to show "true" dimensions on such svg files.
4. The result sprite of these svg files of course differs.
5. I think we're ready to go with resvg. The only thing it's just a tool to convert svg to png. So we need `image-size` package in addition

File: test/svg-sprite.test.js
Comment (Kreeg) [inline test/svg-sprite.test.js:518]: That's the work for prettier. I've tried to install it but it has so many errors even in this file that I  immediately understood why don't you use it here :D
But it strongly recommended anyway.

Comment (XhmikosR) [reply test/svg-sprite.test.js:518]: I don't like prettier and we already enforce lint + style with xo :)

BTW xo has a prettier option, but I'd rather not enable it, at least not until all PRs are merged.

**Comment by XhmikosR** [comment]
OK, so this PR does seem to work AFAICT.

Do you also plan to remove puppeteer from devDependencies later too? Not super-important since it only affects us, just wondering.

After this is merged I say we cut a 2.0.0-beta1 release to test.

**Comment by Kreeg** [comment]
@XhmikosR no, we can't cut puppeteer completely until there is no straight html-to-png package available, like resvg.
I'm saying we can minimize it's influence in tests of course, by comparing svg to it's referencial screenshots by resvg. It means that all screenshots must be updated. Is it OK? If answer is yes I can update this PR lately

**Comment by XhmikosR** [comment]
Yes, I think we can update the screenshots. But let's do it in another PR.

BTW we still use svg2png in grunt-svg-sprite and gulp-svg-sprite, when v2.0.0 is out we can update those too :)

**Comment by Kreeg** [comment]
Ok!

**Review by XhmikosR — APPROVED** [review]

File: test/svg-sprite.test.js
Comment (Kreeg) [reply test/svg-sprite.test.js:518]: I don't like it too. But nevertheless it's a decent tool that automatically does such changes. So I strongly recommend to turn it on.
It's OK to wait until all PR are merged and I have a strategy to use prettier less painfully.
1. ignore all files using `.prettiergnore`
2. Including javascript files step-by-step `.prettiergnore`

Comment (Kreeg) [reply test/svg-sprite.test.js:518]: The effect is not the same. Prettier is more in style formatting. Eslint is more in code quality. This two packages together covering up most cases.
Xo is just `JavaScript/TypeScript linter (ESLint wrapper) with great defaults` as docs says. It has `prettier` integration to ease configuration. I think we really should enforce it to ease core reviewing in future :)

Comment (XhmikosR) [reply test/svg-sprite.test.js:518]: TBH I don't care about prettier *at all*. I use xo and its style and I'm happy with it.

But like I said, we can revisit after all PRs have been addressed.

**Comment by Kreeg** [comment]
Quite strange things happens with tests in windows machines. Not sure why random tests are failing by timeout

**Comment by XhmikosR** [comment]
No need to force push the patches, you can restart the actions :)

BTW you should push your patches to this repo so that it's easier to collaborate.

Now, back to the Windows issue. I have seen it randomly happen in other projects too. I don't think it's related to us, rather the runners have limited resources. You can also try to increase mocha's timeout and see if it helps.

**Comment by XhmikosR** [comment]
@Kreeg puppeteer should be removed from peerDependencies

File: lib/svg-sprite/calculate-svg-dimensions.js
Comment (yisibl) [inline lib/svg-sprite/calculate-svg-dimensions.js:18]: Maybe I can return the width and height of the png file directly in [resvg-js](https://github.com/yisibl/resvg-js). I need your feedback.

Comment (Kreeg) [reply lib/svg-sprite/calculate-svg-dimensions.js:18]: @yisibl what'll be great. But I think this should be a new method.

Comment (yisibl) [reply lib/svg-sprite/calculate-svg-dimensions.js:18]: Okay.

**Pull request merged**
